### ISSUE TITLE
is_successful typecheck fails in 0.7.0

### ISSUE THREAD

**Issue by Heliozoa** [dry-python/returns#93]
Hi,
I faced the following issue in 0.7.0: mypy gives an error when `is_successful` is given a `Result` type. Given the following code (adapted from [the docs](https://returns.readthedocs.io/en/0.7.0/pages/result.html#safe)) in main.py using 0.7.0
```python
from returns.result import safe, is_successful

@safe
def divide(number: int) -> float:
    return number / number

is_successful(divide(1))
```
`mypy main.py` (mypy 0.701) fails with
```bash
main.py:7: error: No overload variant of "is_successful" matches argument type "Result[float, Exception]"
main.py:7: note: Possible overload variants:
main.py:7: note:     def is_successful(container: Success[Any]) -> Literal[True]
main.py:7: note:     def is_successful(container: Failure[Any]) -> Literal[False]
```
The typecheck passes with 0.6.0 (importing the functions from `returns.functions` rather than `returns.result`) and with the `is_successful` examples in the docs where it's given a known `Success` or `Failure`. The type error aside, the function works.
Thanks for your work on the project.

**Comment by sobolevn** [comment]
@Heliozoa yes, there was a bug, which is fixed in `0.8.0`. I have just released it.
Please, make sure to follow the `CHANGELOG`, since there are a lot of changes (including typing).

**Comment by sobolevn** [comment]
Released!

```
 - Uploading returns-0.8.0-py3-none-any.whl 100%
 - Uploading returns-0.8.0.tar.gz 100%
```

There's also a new `mypy` plugin to fix decorator types.
